docs/debugging.rst

```diff
@@ -3,68 +3,142 @@ Debugging
 
-This creates a sandbox for the application with the given ID and, instead
-of running the application, runs a shell inside the sandbox. From the shell
-prompt, it is then possible to run the application. This can also be done
-using any debugging tools that you want to use. For example, to run the
-application with ``gdb``::
+  $ flatpak info --show-sdk $FLATPAK_ID
+  org.freedesktop.Sdk/x86_64/23.08
 
- $ gdb /app/bin/<application-binary>
+Now install the SDK and the Debug package for that SDK::
 
-This works because the ``--devel`` option tells Flatpak to use the SDK as the
-runtime, which includes debugging tools like ``gdb``. The ``--devel`` option
-also adjusts the sandbox setup to enable debugging.
+  $ flatpak install org.freedesktop.Sdk/x86_64/23.08
+  $ flatpak install org.freedesktop.Sdk.Debug/x86_64/23.08
 
-.. note::
+Additionally install the Debug package for the application if available::
 
-  The Freedesktop SDK (on which many others are based), includes a range
-  of debugging tools, such as ``gdb``, ``strace``, ``nm``, ``dbus-send``,
-  ``dconf``, and many others.
+  $ flatpak install $FLATPAK_ID.Debug
 
-``gdb`` is much more useful when it has access to debug information for the
-application and the runtime it is using. Flatpak splits this information off
-into debug extensions, which you should install before debugging an
-application::
+Debug shell
+-----------
 
- $ flatpak install <runtime-id>.Debug
+A debugging environment can be created by starting a shell inside the
+sandbox::
 
-When the ``--devel`` option is used, Flatpak will automatically use any
-matching debug extensions that it finds.
+  $ flatpak run --command=sh --devel --filesystem=$(pwd) $FLATPAK_ID
 
-It is also possible to get a shell inside an application sandbox without having
-to install it. This is done using ``flatpak-builder``'s ``--run`` option::
+This creates a sandbox for the application with the given ID and, instead
+of running the application, runs a shell inside the sandbox. The
+``--devel`` option tells Flatpak to use the SDK as the runtime, which
+includes various debugging tools and it also adjusts the sandbox setup
+to enable debugging.
+
+It is also possible to get a shell inside an application sandbox without
+having to install it. This is done using ``flatpak-builder``'s ``--run``
+option::
 
  $ flatpak-builder --run <build-dir> <manifest> sh
 
 This sets up a sandbox that is populated with the build results found in
 the build directory, and runs a shell inside it.
 
-Creating a .Debug extension
+Using GDB in the sandbox
+------------------------
+
+Note that :ref:`debugging:Debug packages` must be installed to get
+meaningful traces from GDB. Once inside the :ref:`debugging:Debug shell`
+to run the application with ``gdb`` ::
+
+ $ gdb /app/bin/<application-binary>
+
+To pass arguments to the application::
+
+  $ (gdb) run <arguments>
+
+Or alternatively::
+
+  $ gdb --args /app/bin/<application-binary> <arguments>
+  $ (gdb) run
```

Comment:
```suggestion
To pass arguments to the application::

  $ gdb --args /app/bin/<application-binary> <arguments>
  $ (gdb) run
```

For simplicity a single recommendation is fine IMO.

Reply:
Done